surface/include/pcl/surface/grid_projection.h

```diff
@@ -92,7 +94,7 @@ namespace pcl
         Eigen::Vector3f vect_at_grid_pt;
       };
 
-      typedef boost::unordered_map<int, Leaf, boost::hash<int>, std::equal_to<int>, Eigen::aligned_allocator<int> > HashMap;
+      typedef std::unordered_map<int, Leaf, std::hash<int>, std::equal_to<int>, Eigen::aligned_allocator<std::pair<const int, Leaf>>> HashMap;
```

Comment:
This allocator looked suspicious. The convention is to allocated the full object that will be created. The moment it started using `std::unordered_map`, it triggered a static assertion which complained it was not a valid allocator, so now it's changed to conform with the default.